hybridCarboniteMetadataService/src/main/scala/cromwell/services/metadata/hybridcarbonite/CarboniteMetadataServiceActor.scala

```diff
@@ -2,36 +2,62 @@ package cromwell.services.metadata.hybridcarbonite
 
 import akka.actor.{Actor, ActorLogging, ActorRef, Props}
 import cats.data.NonEmptyList
+import cromwell.services.ServiceRegistryActor.{IoActorRef, NoIoActorRefAvailable, RequestIoActorRef}
 import cromwell.services.metadata.MetadataService.{MetadataReadAction, MetadataWriteAction, MetadataWriteFailure}
+import cromwell.services.metadata.hybridcarbonite.CarbonitedMetadataThawingActor.ThawCarboniteFailed
 import cromwell.util.GracefulShutdownHelper
 import cromwell.util.GracefulShutdownHelper.ShutdownCommand
 
+import scala.concurrent.ExecutionContext
+import scala.concurrent.duration._
+
 class CarboniteMetadataServiceActor(carboniteConfig: HybridCarboniteConfig, serviceRegistryActor: ActorRef) extends Actor with ActorLogging with GracefulShutdownHelper {
 
-  // TODO: [CARBONITE] Pass in a real reference to the IO actor somehow
-  val ioActor = context.actorOf(Props(new Actor {
-    override def receive: Receive = Actor.emptyBehavior }))
+  implicit val ec: ExecutionContext = context.dispatcher
+
+  var ioActorOption: Option[ActorRef] = None
+  scheduleIoActorLookup()
 
   // TODO: [CARBONITE] Validate the carbonite config
-  def thawingActorProps(): Props = CarbonitedMetadataThawingActor.props(carboniteConfig, serviceRegistryActor, ioActor)
+  def thawingActorProps(ioActor: ActorRef): Props = CarbonitedMetadataThawingActor.props(carboniteConfig, serviceRegistryActor, ioActor)
 
-  val carboniteWorker: Option[ActorRef] = {
-    carboniteConfig.carboniteInterval map { interval => context.actorOf(CarboniteWorkerActor.props(interval)) }
-  }
+  var carboniteWorker: Option[ActorRef] = None
 
   override def receive: Receive = {
     case read: MetadataReadAction =>
-      val worker = context.actorOf(CarbonitedMetadataReaderActor.props)
-      worker forward read
+      ioActorOption match {
+        case Some(ioActor) =>
+          val worker = context.actorOf(thawingActorProps(ioActor))
+          worker forward read
+        case None =>
+          sender ! ThawCarboniteFailed(new Exception("Cannot create CarbonitedMetadataThawingActor: no IoActor reference available"))
```

Comment:
fail immediately rather than buffer this for a while?

Reply:
This seemed like the easiest solution. 

I *think* I can justify it a little bit by saying we should only get here if someone runs a metadata read within a tiny (microsecond-wide) window between the two actors starting up